.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -40,33 +40,44 @@ This is a checklist for PR authors & reviewers. Please make sure to complete all
-    - [ ] I followed proper naming convention for platform-specific files (if applicable)
-    - [ ] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
-    - [ ] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
+     - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. “toggleReport” and not “onIconClick”)
+    - [ ] I verified that comments were added to code that is not self explanatory
+    - [ ] I verified that any new or modified comments were clear, correct english, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
+    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
+    - [ ] I verified any copy / text that was added to the app is correct english and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
+    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
+    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
+- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
 - [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
 - [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
+- [ ] I verified all code is DRY
+- [ ] I verified any variables that can be defined as constants (ie. in CONST.js) are defined as such
 - [ ] If a new component is created I verified that:
     - [ ] A similar component doesn't exist in the codebase
-    - [ ] Has the `displayName` property if it’s a Functional Component
-    - [ ] Has Storybook stories (optional)
-    - [ ] Has Unit tests (optional)
+    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
+    - [ ] Any functional components have the `displayName` property
+    - [ ] The file is named correctly
+    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
+    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
+    - [ ] Any internal methods are bound to “this” properly so there are no scoping issues
+    - [ ] Any internal methods bound to “this” are necessary to be bound
```

Comment:
These two instructions are not clear to me. Am I supposed to only bind to "this" if we use "this" inside the internal method? or always?